plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -227,4 +237,55 @@ private static String quoteIdentifier(String identifier)
     {
         return format("\"%s\"", identifier.replaceAll("\"", "\"\""));
     }
+
+    public static Optional<String> getQueryOptionsString(String options)
+    {
+        if (isNullOrEmpty(options)) {
+            return Optional.empty();
+        }
+
+        Map<String, String> queryOptionsMap = parseQueryOptions(options);
+        return getQueryOptions(queryOptionsMap);
+    }
+
+    public static Optional<String> getQueryOptions(Map<String, String> queryOptionsMap)
+    {
+        if (queryOptionsMap.isEmpty()) {
+            return Optional.empty();
+        }
+        StringBuilder result = new StringBuilder();
+        for (Map.Entry<String, String> entry : queryOptionsMap.entrySet()) {
+            result.append("SET ")
+                    .append(entry.getKey())
+                    .append(" = ")
+                    .append(entry.getValue())
+                    .append(";\n");
+        }
+        return Optional.of(result.toString());
```

Comment:
Pottentially more concise

```suggestion
        return Optional.of(queryOptionsMap.entrySet().stream()
                .map(e -> "SET " + e.getKey() + " = " + e.getValue())
                .collect(Collectors.joining(";\n")));

```

Reply:
@findinpath done.